File: demo/app_lib/src/main/java/com/bytedance/app_lib/LibraryDetailScene.java
package com.bytedance.app_lib;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.bytedance.scene.ui.template.AppCompatScene;
import com.bytedance.scenerouter.annotation.RouterValue;
import com.bytedance.scenerouter.annotation.SceneUrl;
import com.bytedance.scenerouter.core.SceneRouters;

import java.io.Serializable;
import java.util.ArrayList;

@SceneUrl({"/test1", "/test2", "/test3"})
public class LibraryDetailScene extends AppCompatScene {
    @RouterValue("haha")
    public String var00;

    @RouterValue("char211111")
    public CharSequence var01;

    @RouterValue
    public CharSequence[] var02;

    @RouterValue
    public ArrayList<String> var03;

    @RouterValue
    public String[] var04;

    @RouterValue
    public Bundle var05;

    @RouterValue
    public ArrayList<CharSequence> var06;

    @RouterValue
    public ArrayList<Integer> var07;

    @RouterValue
    public Uri var08;

    @RouterValue
    public FFF var09;

    @RouterValue
    public Uri[] var101;

    @RouterValue
    public int var0;

    @RouterValue
    public long var1;

    @RouterValue
    public char var2;

    @RouterValue
    public float var3;

    @RouterValue
    public short var4;

    @RouterValue
    public boolean var5;

    @RouterValue
    public byte var6;

    @RouterValue
    public double var7;

    @RouterValue
    public int[] var8;

    @RouterValue
    public long[] var9;

    @RouterValue
    public char[] var10;

    @RouterValue
    public float[] var11;

    @RouterValue
    public short[] var12;

    @RouterValue
    public boolean[] var13;

    @RouterValue
    public byte[] var14;

    @RouterValue
    public double[] var15;

    @RouterValue
    public ArrayList<Uri> var16;

    @NonNull
    @Override
    public View onCreateContentView(@NonNull LayoutInflater inflater, @NonNull ViewGroup container, @Nullable Bundle savedInstanceState) {
        return new View(requireSceneContext());
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        setTitle("Library Scene");
        SceneRouters.bind(this);
        Toast.makeText(getSceneContext(), var00, Toast.LENGTH_SHORT).show();
    }

    public static class FFF implements Serializable {

    }
}

File: app_lib_1/src/main/java/org/qiibeta/lib1/LibraryDetailScene.kt
package org.qiibeta.lib1

import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.bytedance.scene.ui.template.AppCompatScene
import com.bytedance.scenerouter.annotation.RouterValue
import com.bytedance.scenerouter.annotation.SceneUrl
import com.bytedance.scenerouter.core.SceneRouters
import java.io.Serializable
import java.util.*


@SceneUrl("/test1", "/test2", "/test3")
class LibraryDetailScene : AppCompatScene() {
    @JvmField
    @RouterValue("haha")
    var var00: String? = null

    @JvmField
    @RouterValue("char211111")
    var var01: CharSequence? = null

    @JvmField
    @RouterValue
    var var02: Array<CharSequence>? = null

    @JvmField
    @RouterValue
    var var03: ArrayList<String>? = null

    @JvmField
    @RouterValue
    var var04: Array<String>? = null

    @JvmField
    @RouterValue
    var var05: Bundle? = null

    @JvmField
    @RouterValue
    var var06: ArrayList<CharSequence>? = null

    @JvmField
    @RouterValue
    var var07: ArrayList<Int>? = null

    @JvmField
    @RouterValue
    var var08: Uri? = null

    @JvmField
    @RouterValue
    var var09: FFF? = null

    @JvmField
    @RouterValue
    var var101: Array<Uri>? = null

    @JvmField
    @RouterValue
    var var0 = 0

    @JvmField
    @RouterValue
    var var1: Long = 0

    @JvmField
    @RouterValue
    var var2 = 0.toChar()

    @JvmField
    @RouterValue
    var var3 = 0f

    @JvmField
    @RouterValue
    var var4: Short = 0

    @JvmField
    @RouterValue
    var var5 = false

    @JvmField
    @RouterValue
    var var6: Byte = 0

    @JvmField
    @RouterValue
    var var7 = 0.0

    @JvmField
    @RouterValue
    var var8: IntArray? = null

    @JvmField
    @RouterValue
    var var9: LongArray? = null

    @JvmField
    @RouterValue
    var var10: CharArray? = null

    @JvmField
    @RouterValue
    var var11: FloatArray? = null

    @JvmField
    @RouterValue
    var var12: ShortArray? = null

    @JvmField
    @RouterValue
    var var13: BooleanArray? = null

    @JvmField
    @RouterValue
    var var14: ByteArray? = null

    @JvmField
    @RouterValue
    var var15: DoubleArray? = null

    @JvmField
    @RouterValue
    var var16: ArrayList<Uri>? = null

    override fun onCreateContentView(
        inflater: LayoutInflater,
        container: ViewGroup,
        savedInstanceState: Bundle?
    ): View {
        return View(requireSceneContext())
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        setTitle("Library Scene")
        SceneRouters.bind(this)
        Toast.makeText(getSceneContext(), var00, Toast.LENGTH_SHORT).show()
    }

    class FFF : Serializable
}
